packages/codegen-ui-react/lib/utils/forms/value-mappers.ts

```diff
@@ -0,0 +1,360 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+import { factory, NodeFlags, SyntaxKind } from 'typescript';
+
+/**
+ * accepts timestamp number and returns date object
+ * since datastore allows second and millisecond epochs we need to format accordingly
+ *
+ * @param ts number
+ * @returns date object
+ */
+export const convertTimeStampToDate = (ts: number): Date => {
+  if (Math.abs(Date.now() - ts) < Math.abs(Date.now() - ts * 1000)) {
+    return new Date(ts);
+  }
+  return new Date(ts * 1000);
+};
+
+/**
+ * takes a date object and converts it to the local format used for an html input element with datetime-local
+ *
+ * @param date date object
+ * @returns string formatted in datetime-local
+ */
+export const convertToLocal = (date: Date): string => {
+  const df = new Intl.DateTimeFormat('default', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+    fractionalSecondDigits: 3,
+  });
+  const parts = df.formatToParts(date).map((v) => v.value);
+  return `${parts[4]}-${parts[0]}-${parts[2]}T${parts[6]}:${parts[8]}:${parts[10]}.${parts[12]}`;
```

Comment:
This will not work for PM, `df.formatToParts(new Date('Mon Sep 26 2022 20:55:58 GMT-0700 (Pacific Daylight Time)'))` gives `parts[8] === "08"` instead of `"20"`

Reply:
Adding `hour12: false` to the config object above will address the AM/PM issue but another thing to be aware of is that the order of parts also changes depending on the locale used by DateTimeFormat. Not sure if `'default'` is meant to vary by the user's actual locale or not but that could work against us for this implementation.

Reply:
new rev
- changing to set hour12 to false
- for the times we can just index by the key as we know thats a constant for the response from date time format